### PR TITLE
CLOUD-2066. Add CCM client to CB images.

### PR DESCRIPTION
**Pull request by jadair-cloudera**
With these changes, the CCM client infrastructure and prerequisites
have been added to the Salt configuration for CB images.

### CODE REVIEW

**Comment by jadair-cloudera** [comment]
FYI, here is the client package module based on the code that is currently running in Altus. I renamed some of the scripts for consistency and got everything into Salt, but there are no functional changes.
https://github.infra.cloudera.com/thunderhead/thunderhead/tree/master/clients/autossh-adventure

File: saltstack/base/salt/ccm-client/cdp/bin/update-reverse-tunnel-values.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+CCM_SSH_PORT=${CCM_SSH_PORT}
+HOST_PORT=${HOST_PORT}
+ROLE=${ROLE}
+HOST=${HOST}
+TUNNEL_INITIATOR_ID=${TUNNEL_INITIATOR_ID}
+ENCIPHERED_PRIVATE_KEY=${ENCIPHERED_PRIVATE_KEY}
+CLOUD_PROVIDER=${CLOUD_PROVIDER}
+PUBLIC_KEY=${PUBLIC_KEY}
+EOF
+systemctl enable tunnel@{ROLE}.service
```
Comment (pdarvasi) [inline saltstack/base/salt/ccm-client/cdp/bin/update-reverse-tunnel-values.sh]: This should be done via salt: https://docs.saltstack.com/en/latest/ref/states/all/salt.states.service.html#salt.states.service.running

File: saltstack/base/salt/ccm-client/init.sls [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  file.managed:
+    - name: /cdp/bin/reverse-tunnel-values.sh
+    - source: salt://{{ slspath }}/cdp/bin/reverse-tunnel-values.sh
+
+/cdp/bin/update-reverse-tunnel-values.sh:
+  file.managed:
+    - name: /cdp/bin/update-reverse-tunnel-values.sh
+    - source: salt://{{ slspath }}/cdp/bin/updatereverse-tunnel-values.sh
+
+/systemd/tunnel@.service:
```
Comment (pdarvasi) [inline saltstack/base/salt/ccm-client/init.sls]: We should enable the service IMHO here: https://docs.saltstack.com/en/latest/ref/states/all/salt.states.service.html#salt.states.service.running

File: saltstack/base/salt/prerequisites/packages.sls [hunk 3]
```diff
@@ -43,6 +43,9 @@ packages_install:
   {% endif %}
       - deltarpm
       - nvme-cli
+      - autossh
+      - openssl
+      - vim-common
```
Comment (pdarvasi) [inline saltstack/base/salt/prerequisites/packages.sls]: Is this one really needed?

File: saltstack/base/salt/ccm-client/cdp/bin/reverse-tunnel.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+export AUTOSSH_LOGLEVEL=7
+export AUTOSSH_LOGFILE="/var/log/autossh-${ROLE}.log"
+# Set max backoff interval in case of recurring connection failures.
+export AUTOSSH_POLL=30
+
+PRIVATE_KEY=/tmp/pk.key
+
+if grep -q "BEGIN" ${ENCIPHERED_PRIVATE_KEY}; then
+    cat ${ENCIPHERED_PRIVATE_KEY} > ${PRIVATE_KEY}
+else
```
Comment (pdarvasi) [inline saltstack/base/salt/ccm-client/cdp/bin/reverse-tunnel.sh]: Do we have such a use-case when there is unencrypted ENCIPHERED_PRIVATE_KEY?

Comment (spacharya) [reply saltstack/base/salt/ccm-client/cdp/bin/reverse-tunnel.sh]: Well technically in all production usages, we should be using Enciphered value, But in testing and debugging, depending on the team doing the work, they might want to use the plain key as is. That was the usecase. Also, Altus originally supported unencrypted and for backward incompatibility, we kept it as is.

File: saltstack/base/salt/prerequisites/packages.sls (hunk 3 above)
Comment (spacharya) [reply saltstack/base/salt/prerequisites/packages.sls]: `$(xxd -pu <<< $(echo ${TUNNEL_INITIATOR_ID} |`

vim-commons provides the xxd funtionality which is used to decipher the enciphered text. If the xxd package is natively present, (i believe it is on ubunutu not on centos) then you can skip it. But for consistency, you might want to keep it around.

File: saltstack/base/salt/ccm-client/cdp/bin/reverse-tunnel.sh (hunk 4 above)
Comment (jadair-cloudera) [reply saltstack/base/salt/ccm-client/cdp/bin/reverse-tunnel.sh]: Peter, I will resolve this conversation. Reopen if you still have concerns.

File: saltstack/base/salt/prerequisites/packages.sls (hunk 3 above)
Comment (jadair-cloudera) [reply saltstack/base/salt/prerequisites/packages.sls]: Peter, I will resolve this conversation. Reopen if you still have concerns.

Comment (pdarvasi) [reply saltstack/base/salt/prerequisites/packages.sls]: Ok, got it, thanks

**Pull request merged**